src/RoomDetail/components/RoomCalendar.tsx

```diff
@@ -1,64 +1,57 @@
+import { useEffect, useRef, useContext } from 'react';
 import clsx from 'clsx';
+import { makeDays } from '../utils/utils';
+import { DAY_OF_THE_WEEK } from '../constants/constant';
 import { Icon } from '@/shared/Icon';
+import { DateRoomDetailContext } from '@/pages/RoomDetailPage';
 
-const calendar = [
-  {
-    day: '월',
-    date: '8',
-    bug: true,
-    point: false
-  },
-  {
-    day: '화',
-    date: '9',
-    bug: true,
-    point: false
-  },
-  {
-    day: '수',
-    date: '10',
-    bug: false,
-    point: false
-  },
-  {
-    day: '목',
-    date: '11',
-    bug: true,
-    point: true
-  },
-  {
-    day: '금',
-    date: '12',
-    bug: true,
-    point: false
-  },
-  {
-    day: '토',
-    date: '13',
-    bug: false,
-    point: false
-  },
-  {
-    day: '일',
-    date: '14',
-    bug: true,
-    point: false
-  }
-];
+interface RoomCalendarProps {
+  certifiedDates: string[];
+}
+
+const RoomCalendar = ({ certifiedDates }: RoomCalendarProps) => {
+  const dateRef = useRef<HTMLDivElement>(null);
+  const { changeDate, date: changedDate } = useContext(DateRoomDetailContext);
+
+  const { thisWeekDates, today } = makeDays();
+  const stringThisWeekDates = thisWeekDates.map(
+    (el) => `${el.getFullYear()}-${el.getMonth() + 1}-${el.getDate()}`
+  );
+
+  useEffect(() => {
+    const handleOutsideClose = (e: MouseEvent) => {
+      if (dateRef.current && !dateRef.current?.contains(e.target as Node)) {
+        changeDate(new Date());
+      }
+    };
+    document.addEventListener('click', handleOutsideClose);
+
+    return () => document.removeEventListener('click', handleOutsideClose);
+  }, [changeDate]);
```

Comment:
ASK:
`useEffect` 를 사용하지 않더라도 닫는 이벤트를 처리할 수 있을 것 같은데요!
혹시 아래와 같은 느낌으로 작성한다면 문제가 발생할지 궁금해요!

```suggestion
  const handleClose = (e: MouseEvent) => {
    if (e.target !== dateRef.current) {
      return;
    }

    changeDate(new Date());
  };
```

```tsx
      <div
        className="flex justify-between"
        ref={dateRef}
        onClick={handleClose}
      >
```

![image](https://github.com/team-moabam/moabam-FE/assets/50488780/ed1bd93e-5f7a-4606-b7c9-ad29b3593d74)


https://ko.react.dev/learn/you-might-not-need-an-effect